api_tests/base/test_parsers.py

```diff
@@ -0,0 +1,120 @@
+from nose.tools import *  # noqa (PEP8 asserts)
+
+from api.base.parsers import JSONAPIMultipleRelationshipsParser
+
+
+SINGLE_RELATIONSHIP = {
+    'region': {
+        'data': {
+            'type': 'regions',
+            'id': 'us-1'
+        }
+    }
+}
+
+MULTIPLE_RELATIONSHIP = {
+    'affiliated_institutions': {
+        'data': [
+            {
+                'type': 'institutions',
+                'id': 'cos'
+            }, {
+                'type': 'institutions',
+                'id': 'ljaf'
+            }
+        ]
+    }
+}
+
+MIXED_RELATIONSHIP = {
+    'region': {
+        'data': {
+            'type': 'regions',
+            'id': 'us-1'
+        }
+    },
+    'affiliated_institutions': {
+        'data': [
+            {
+                'type': 'institutions',
+                'id': 'cos'
+            }, {
+                'type': 'institutions',
+                'id': 'ljaf'
+            }
+        ]
+    }
+}
+
+MULTIPLE_SINGLE_RELATIONSHIPS = {
+    'node': {
+        'data': {
+            'type': 'nodes',
+            'id': 'abcde'
+        }
+    },
+    'provider': {
+        'data': {
+            'type': 'preprint_providers',
+            'id': 'agrixiv'
+        }
+    }
+}
+
+MULTIPLE_MULTIPLE_RELATIONSHIPS = {
+    'affiliated_institutions': {
+        'data': [
+            {
+                'type': 'institutions',
+                'id': 'cos'
+            }, {
+                'type': 'institutions',
+                'id': 'ljaf'
+            }
+        ]
+    },
+    'providers': {
+        'data': [
+            {
+                'type': 'preprint_providers',
+                'id': 'agrixiv'
+            }, {
+                'type': 'preprint_providers',
+                'id': 'osfpreprints'
+            }
+        ]
+    }
+}
+
+
+class TestMultipleRelationshipsParser:
+
+    def test_flatten_relationships(self):
+
+        parser = JSONAPIMultipleRelationshipsParser()
+
+        # test_single
+        assert_equal(
+            JSONAPIMultipleRelationshipsParser.flatten_relationships(parser, SINGLE_RELATIONSHIP),
+            {'region': 'us-1'}
+        )
+        # test_multiple
+        assert_equal(
+            JSONAPIMultipleRelationshipsParser.flatten_relationships(parser, MULTIPLE_RELATIONSHIP),
```

Comment:
Minor: This test looks like a good candidate for using pytest's parametrization feature:

```python

@pytest.mark.parametrize('relationship,expected',
[
    (SINGLE_RELATIONSHIP, {'region': 'us-1'}),
    (MULTIPLE_RELATIONSHIP, {'affiliated_institutions': ['cos', 'ljaf']}),
])
def test...(relationship, expected):
    parser = ...
    assert JSONBlahblah.flatten_relationships(parser, relationship) == expected
```

Reply:
Ooooooooh that is nifty.